src/render/tutorials/configurable-api-gateway/2_0.html.md

```diff
@@ -270,55 +287,56 @@ public class PaymentsActionFactory implements ActionFactory {
   @Override
   public Action create(String alias, JsonObject config, Vertx vertx, Action doAction) {
     return (fragmentContext, resultHandler) ->
-        Single.just(fragmentContext.getFragment())
-            .map(fragment -> {
-              JsonObject payload = fragment.getPayload();
-              JsonObject user = payload.getJsonObject("user");
-              JsonObject payments = processProviders(payload);
-              fragment.clearPayload();
-              fragment.mergeInPayload(new JsonObject().put(getAlias(alias), payments)
-                  .put("user", user));
-              return new FragmentResult(fragment, FragmentResult.SUCCESS_TRANSITION);
-            })
-            .subscribe(onSuccess -> {
-              Future<FragmentResult> resultFuture = Future.succeededFuture(onSuccess);
-              resultFuture.setHandler(resultHandler);
-            }, onError -> {
-              Future<FragmentResult> resultFuture = Future.failedFuture(onError);
-              resultFuture.setHandler(resultHandler);
-            });
+      Single.just(fragmentContext.getFragment())
+        .map(fragment -> {
+          JsonObject payload = fragment.getPayload();
+          JsonObject user = payload.getJsonObject("user");
+          JsonObject payments = processProviders(payload);
+          fragment.clearPayload();
+          fragment.mergeInPayload(new JsonObject().put(getAlias(alias), payments)
+            .put("user", user));
+          return new FragmentResult(fragment, FragmentResult.SUCCESS_TRANSITION);
+        })
+        .subscribe(onSuccess -> {
+          Future<FragmentResult> resultFuture = Future.succeededFuture(onSuccess);
+          resultFuture.setHandler(resultHandler);
+        }, onError -> {
+          Future<FragmentResult> resultFuture = Future.failedFuture(onError);
+          resultFuture.setHandler(resultHandler);
+        });
   }
 
   private JsonObject processProviders(JsonObject payload) {
     return new JsonObject()
-        .put("timestamp", System.currentTimeMillis())
-        .put("providers", calculateProviders(payload));
+      .put("timestamp", System.currentTimeMillis())
+      .put("providers", PaymentProvidersProcessor.processProviders(payload));
   }
 
   private String getAlias(String alias) {
     return StringUtils.defaultString(alias, "payments");
   }
 }
+
 ```
 
-*src/main/java/io/knotx/example/payment/utils/ProvidersProvider.java*
+*src/main/java/io/knotx/example/payment/utils/PaymentProvidersProcessor.java*
 ```java
 package io.knotx.example.payment.utils;
 
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
 
-public final class ProvidersProvider {
+public final class PaymentProvidersProcessor {
```

Comment:
I liked this `tricky` name ;) Sure, change it.

Reply:
Can you please apply those changes in the example code.

Reply:
I will create dedicated PR for example code and will share PR link shortly.